File: app/src/main/java/me/blog/korn123/easydiary/googledrive/GoogleDriveDownloader.java
package me.blog.korn123.easydiary.googledrive;

import android.app.AlarmManager;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.content.IntentSender;
import android.os.Bundle;

import com.google.android.gms.common.api.ResultCallback;
import com.google.android.gms.drive.Drive;
import com.google.android.gms.drive.DriveApi;
import com.google.android.gms.drive.DriveFile;
import com.google.android.gms.drive.DriveId;
import com.google.android.gms.drive.OpenFileActivityBuilder;

import org.apache.commons.io.IOUtils;

import java.io.FileOutputStream;
import java.io.InputStream;
import java.io.OutputStream;

import me.blog.korn123.commons.constants.Constants;
import me.blog.korn123.commons.utils.CommonUtils;
import me.blog.korn123.commons.utils.EasyDiaryUtils;
import me.blog.korn123.easydiary.helper.EasyDiaryDbHelper;
import me.blog.korn123.easydiary.activities.DiaryMainActivity;

/**
 * Created by CHO HANJOONG on 2016-09-28.
 */
public class GoogleDriveDownloader extends GoogleDriveUtils {

    @Override
    public void onConnected(Bundle connectionHint) {
        super.onConnected(connectionHint);
        // If there is a selected file, open its contents.
        if (mSelectedFileDriveId != null) {
            open();
            return;
        }

        IntentSender intentSender = Drive.DriveApi
                .newOpenFileActivityBuilder()
                .setMimeType(EasyDiaryUtils.getEasyDiaryMimeTypeAll())
                .build(getGoogleApiClient());

        try {
            startIntentSenderForResult(
                    intentSender, REQUEST_CODE_GOOGLE_DRIVE_DOWNLOAD, null, 0, 0, 0);
        } catch (IntentSender.SendIntentException e) {
            e.printStackTrace();
        }
    }

    private void open() {
        DriveFile driveFile = mSelectedFileDriveId.asDriveFile();
        DriveFile.DownloadProgressListener listener = new DriveFile.DownloadProgressListener() {
            @Override
            public void onProgress(long bytesDownloaded, long bytesExpected) {
                // Update progress dialog with the latest progress.
                int progress = (int)(bytesDownloaded*100/bytesExpected);
            }
        };
        driveFile.open(getGoogleApiClient(), DriveFile.MODE_READ_ONLY, listener)
                .setResultCallback(driveContentsCallback);
    }

    private ResultCallback<DriveApi.DriveContentsResult> driveContentsCallback =
        new ResultCallback<DriveApi.DriveContentsResult>() {
            @Override
            public void onResult(DriveApi.DriveContentsResult result) {
                if (result.getStatus().isSuccess()) {
                    try {
                        InputStream driveContents = result.getDriveContents().getInputStream();
                        OutputStream outputStream = new FileOutputStream(EasyDiaryDbHelper.getRealmInstance().getPath());
                        IOUtils.copy(driveContents, outputStream);
                        IOUtils.closeQuietly(outputStream);
                    } catch (Exception e) {
                        e.printStackTrace();
                    }
                }
                Context context = GoogleDriveDownloader.this;
                Intent readDiaryIntent = new Intent(context, DiaryMainActivity.class);
                readDiaryIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP|Intent.FLAG_ACTIVITY_CLEAR_TASK|Intent.FLAG_ACTIVITY_NEW_TASK);
                int mPendingIntentId = 123456;
                PendingIntent mPendingIntent = PendingIntent.getActivity(context, mPendingIntentId, readDiaryIntent, PendingIntent.FLAG_CANCEL_CURRENT);
                AlarmManager mgr = (AlarmManager)context.getSystemService(Context.ALARM_SERVICE);
                mgr.set(AlarmManager.RTC, System.currentTimeMillis() + 100, mPendingIntent);
                System.exit(0);
            }
        };

}

File: app/src/main/java/com/google/android/gms/drive/GoogleDriveDownloader.kt
package com.google.android.gms.drive

import android.app.AlarmManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.content.IntentSender
import android.os.Bundle
import com.google.android.gms.common.api.ResultCallback
import me.blog.korn123.commons.utils.EasyDiaryUtils
import me.blog.korn123.easydiary.activities.DiaryMainActivity
import me.blog.korn123.easydiary.helper.EasyDiaryDbHelper
import org.apache.commons.io.IOUtils
import java.io.FileOutputStream

/**
 * Created by Administrator on 2017-11-21.
 */

class GoogleDriveDownloader : GoogleDriveUtils() {

    override fun onConnected(connectionHint: Bundle?) {
        super.onConnected(connectionHint)
        // If there is a selected file, open its contents.
        if (mSelectedFileDriveId != null) {
            open()
            return
        }

        val intentSender = Drive.DriveApi
                .newOpenFileActivityBuilder()
                .setMimeType(EasyDiaryUtils.getEasyDiaryMimeTypeAll())
                .build(getGoogleApiClient())

        try {
            startIntentSenderForResult(
                    intentSender, REQUEST_CODE_GOOGLE_DRIVE_DOWNLOAD, null, 0, 0, 0)
        } catch (e: IntentSender.SendIntentException) {
            e.printStackTrace()
        }
    }

    private fun open() {
        val driveFile = mSelectedFileDriveId?.asDriveFile()
        val listener = DriveFile.DownloadProgressListener { bytesDownloaded, bytesExpected ->
            // Update progress dialog with the latest progress.
            val progress = (bytesDownloaded * 100 / bytesExpected).toInt()
        }
        driveFile?.open(getGoogleApiClient(), DriveFile.MODE_READ_ONLY, listener)?.setResultCallback(driveContentsCallback)
    }

    private val driveContentsCallback = ResultCallback<DriveApi.DriveContentsResult> { result ->
        if (result.status.isSuccess) {
            try {
                val driveContents = result.driveContents.inputStream
                val outputStream = FileOutputStream(EasyDiaryDbHelper.getRealmInstance().getPath())
                IOUtils.copy(driveContents, outputStream)
                IOUtils.closeQuietly(outputStream)
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }

        val context = this@GoogleDriveDownloader
        val readDiaryIntent = Intent(context, DiaryMainActivity::class.java)
        readDiaryIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_NEW_TASK)
        val mPendingIntentId = 123456
        val mPendingIntent = PendingIntent.getActivity(context, mPendingIntentId, readDiaryIntent, PendingIntent.FLAG_CANCEL_CURRENT)
        val mgr = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager
        mgr.set(AlarmManager.RTC, System.currentTimeMillis() + 100, mPendingIntent)
        System.exit(0)
    }
}